File: Chapter2.py
import cv2
import numpy as np

#Grayscale# #Blur Canny Dilate Erode#

img = cv2.imread("lena.png")
imgGray = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
imgBlur = cv2.GaussianBlur(imgGray,(7,7),0)
imgCanny = cv2.Canny(imgBlur,75,75)

kernel = np.ones((5, 5), np.uint8)

imgDilate = cv2.dilate(imgCanny, kernel, iterations=1)
imgErode = cv2.erode(imgDilate,kernel,iterations=1)

cv2.imshow("Original Image", img)
cv2.imshow("Gray Image", imgGray)
cv2.imshow("Blur Image", imgBlur)
cv2.imshow("Canny Image", imgCanny)
cv2.imshow("Dilate Image", imgDilate)
cv2.imshow("Erode Image", imgErode)

cv2.waitKey(0)

File: Chapter 2.cpp
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>
#include <iostream>

using namespace cv;
using namespace std;


//Color---- Grayscale//

//int main() {
//
//	string path = "Resources/test.png";
//	Mat img = imread(path);
//	Mat imgGray;
//	cvtColor(img, imgGray, COLOR_BGR2GRAY);
//	imshow("Image",img);
//	imshow("Image Gray",imgGray);
//	waitKey(0);
//
//	return 0;
//}


//Adding Blur//

//int main() {
//	
//		string path = "Resources/test.png";
//		Mat img = imread(path);
//		Mat imgBlur;
//		GaussianBlur(img, imgBlur, Size(7,7), 5, 0);
//		imshow("Image",img);
//		imshow("Image Blur",imgBlur);
//		waitKey(0);
//	
//		return 0;
//	}


//Edge detector//
//int main() {
//
//	string path = "Resources/test.png";
//	Mat img = imread(path);
//	Mat imgBlur, imgCanny;
//	GaussianBlur(img, imgBlur, Size(7, 7), 5, 0);  //Before Canny, a small amount of Blur is good practice//
//	Canny(img, imgCanny, 50, 150);
//	imshow("Image", img);
//	imshow("Image Blur", imgBlur);
//	imshow("Image Canny", imgCanny);
//	waitKey(0);
//
//	return 0;
//}
//


//Edge Dilation/Erosion//
int main() {

	string path = "Resources/test.png";
	Mat img = imread(path);
	Mat imgBlur, imgCanny, imgDil, imgErode;
	GaussianBlur(img, imgBlur, Size(7, 7), 5, 0);  //Before Canny, a small amount of Blur is good practice//
	Canny(img, imgCanny, 50, 150);
	Mat kernel = getStructuringElement(MORPH_RECT, Size(3, 3));
	dilate(imgCanny, imgDil, kernel);
	erode(imgDil, imgErode, kernel);
	imshow("Image", img);
	imshow("Image Blur", imgBlur);
	imshow("Image Canny", imgCanny);
	imshow("Image Dilation", imgDil);
	imshow("Image Erosion", imgErode);
	waitKey(0);

	return 0;
}



